lib/metalsmith.js

```diff
@@ -22,12 +23,42 @@ const views = [
   paths.govukfrontend
 ]
 
+const coloursAppliedSCSS = scssToJson('./node_modules/@govuk-frontend/globals/settings/_colours-applied.scss', {
+  dependencies: [{path: './node_modules/@govuk-frontend/globals/settings/_colours-palette.scss'}]
+})
+
+const coloursPaletteSCSS = scssToJson('./node_modules/@govuk-frontend/globals/settings/_colours-palette.scss')
+
+// combine both SCSS data objects - some Main colours are only in Palette
+
+const coloursSCSS = Object.assign(coloursAppliedSCSS, coloursPaletteSCSS)
+
+// process colours data
+
+var colours = require('../data/colours.json')
+
+for (var groupName in colours.main){
+  var group = colours.main[groupName]
+  for (var colour of group){
+    colour.colour = coloursSCSS[colour.name]
+  }
+}
+
+for (var colour of colours.greyscale){
+  colour.colour = coloursSCSS[colour.name]
+}
+
+for (var colour of colours.extended){
+  colour.colour = coloursSCSS[colour.name]
+}
+
```

Comment:
I wonder if the colour processing should move to its own file, the other stuff in metalsmith.js feels quite high level whereas this is specific to a page.